plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/workspace/WorkspaceProjectStorageCleanerImpl.java

```diff
@@ -0,0 +1,96 @@
+ * from Codenvy S.A..
+ */
+package com.codenvy.workspace;
+
+import com.codenvy.machine.backup.WorkspaceIdHashLocationFinder;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.che.api.core.util.CommandLine;
+import org.eclipse.che.api.core.util.ListLineConsumer;
+import org.eclipse.che.api.core.util.ProcessUtil;
+import org.eclipse.che.api.workspace.server.WorkspaceProjectStorageCleaner;
+import org.eclipse.che.commons.lang.concurrent.ThreadLocalPropagateContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Component to clean up workspace project storage which can be located in the remote node.
+ *
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class WorkspaceProjectStorageCleanerImpl implements WorkspaceProjectStorageCleaner {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WorkspaceProjectStorageCleanerImpl.class);
+
+    private final WorkspaceIdHashLocationFinder workspaceIdHashLocationFinder;
+    private final File                          backupsRootDir;
+    private final int                           cleanUpTimeOut;
+    private final String                        workspaceCleanUpScript;
+    private final ExecutorService               executor;
+
+    @Inject
+    public WorkspaceProjectStorageCleanerImpl(WorkspaceIdHashLocationFinder workspaceIdHashLocationFinder,
+                                              @Named("che.user.workspaces.storage") File backupsRootDir,
+                                              @Named("workspace.cleanup.cleanup_workspace_storage_script") String workspaceCleanUpScript,
+                                              @Named("workspace.cleanup.cleanup_duration_second") int cleanUpTimeOut) {
+        this.workspaceIdHashLocationFinder = workspaceIdHashLocationFinder;
+        this.backupsRootDir = backupsRootDir;
+        this.workspaceCleanUpScript = workspaceCleanUpScript;
+        this.cleanUpTimeOut = cleanUpTimeOut;
+        executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("WorkspaceProjectStorageCleanerImpl-%d")
+                                                                           .setDaemon(true)
+                                                                           .build());
+    }
+
+    @Override
+    public void remove(String workspaceId) {
+        File backupWorkspaceFolder = workspaceIdHashLocationFinder.calculateDirPath(backupsRootDir, workspaceId);
+
+        CommandLine commandLine = new CommandLine(workspaceCleanUpScript, backupWorkspaceFolder.getAbsolutePath());
```

Comment:
And we need new one property with name remote host... For now we have not such one.